applications/FSIapplication/python_scripts/fsi_analysis.py

```diff
@@ -77,17 +77,52 @@ def Initialize(self):
         self.ModifyInitialProperties()
         self.ModifyInitialGeometry()
 
-        self._GetListOfProcesses()
-        self._SetUpAnalysis()
+        # Initialize the user-provided processes
+        self._AnalysisStage__CreateListOfProcesses() # Why name mangling?
+        for process in self._GetListOfProcesses():
+            process.ExecuteInitialize()
+
+        self._GetSolver().Initialize()
+        self.Check()
+
+        self.ModifyAfterSolverInitialize()
 
         for process in self._GetListOfProcesses():
             process.ExecuteBeforeSolutionLoop()
 
+        ## Stepping and time settings
```

Comment:
If this stuff would be in a separate function then you could reuse more things, I will keep it in mind